src/pages/timecard/Timecard.jsx

```diff
@@ -6,20 +6,53 @@ import BackLink from '../../components/common/form/navigation/backlink/BackLink'
 import SelectTimecardPeriodType from '../../components/timecard/select-timecard-period-type/SelectTimecardPeriodType';
 import ErrorSummary from '../../components/common/form/error-summary/ErrorSummary';
 import generateDocumentTitle from '../../utils/generate-document-title/generateDocumentTitle';
+import { getTimeEntries } from '../../api/services/timecardService';
+import { formatTime } from '../../utils/time-entry-utils/timeEntryUtils';
+import { UrlSearchParamBuilder } from '../../utils/api-utils/UrlSearchParamBuilder';
 import EditShiftTimecard from '../../components/timecard/edit-shift-timecard/EditShiftTimecard';
 import { useTimecardContext } from '../../context/TimecardContext';
+import { useApplicationContext } from '../../context/ApplicationContext';
 
 import { sortErrorKeys } from '../../utils/sort-errors/sortErrors';
 
+const updateTimeEntryContextData = async (setTimeEntries) => {
+  const { timePeriodTypes } = useApplicationContext();
+  const timeEntriesParams = new UrlSearchParamBuilder()
+    .setTenantId('00000000-0000-0000-0000-000000000000')
+    .setFilter('ownerId==1')
+    .getUrlSearchParams();
+  const timeEntriesResponse = await getTimeEntries(timeEntriesParams);
+
+  if (timeEntriesResponse.data.items?.length > 0) {
+    const existingTimeEntries = await Promise.all(
+      timeEntriesResponse.data.items.map(async (timeEntry) => {
+        console.log(timePeriodTypes[timeEntry.timePeriodTypeId]);
+        return {
+          timeEntryId: timeEntry.id,
+          timePeriodType: timePeriodTypes[timeEntry.timePeriodTypeId],
+          startTime: formatTime(timeEntry.actualStartTime),
+          finishTime: timeEntry.actualEndTime
+            ? formatTime(timeEntry.actualEndTime)
+            : '',
+          timePeriodTypeId: timeEntry.timePeriodTypeId,
+        };
+      })
+    );
```

Comment:
I think I’m right in saying that this doesn’t need to be async anymore?

```suggestion
    const existingTimeEntries = timeEntriesResponse.data.items.map(
      (timeEntry) => ({
        timeEntryId: timeEntry.id,
        timePeriodType: timePeriodTypes[timeEntry.timePeriodTypeId],
        startTime: formatTime(timeEntry.actualStartTime),
        finishTime: timeEntry.actualEndTime
          ? formatTime(timeEntry.actualEndTime)
          : '',
        timePeriodTypeId: timeEntry.timePeriodTypeId,
      })
    );
```

Reply:
You're right! Good catch, removed the async

Reply:
Ah well there’s that and also that you don’t need the `await` and the `Promise.all`. In my original comment there’s a code suggestion that strips this out.

Reply:
I've done this now